urldownloader.py

```diff
@@ -125,6 +125,149 @@ def __init__(self, config=None) -> None:
+            yaml.dump(kangooroo_config, temp_conf)
+
+
+        # Set up environment variable and commandline arguments for running kangooroo
+        env_variables = {
+            "JAVA_OPTS": f"-Xmx{math.floor(self.service_attributes.docker_config.ram_mb*0.75)}m"
+        }
+        kangooroo_args = [
+                "./bin/kangooroo",
+                "--conf-file",
+                temp_conf.name,
+                "-mods",
+                "summary,captcha",
+                "--simple-result",
+                "--url",
+                request.task.fileinfo.uri_info.uri,
+        ]
+        if self.no_sandbox:
+            kangooroo_args.insert(-2, "--no-sandbox")
+        try:
+            subprocess.run(kangooroo_args, cwd=KANGOOROO_FOLDER,  timeout=self.request_timeout, env=env_variables)
+        except subprocess.TimeoutExpired:
+            timeout_section = ResultTextSection("Request timed out", parent=request.result)
+            timeout_section.add_line(
+                f"Timeout of {self.request_timeout} seconds was not enough to process the query fully."
+            )
+            return None, None
+
+        url_md5 = hashlib.md5(request.task.fileinfo.uri_info.uri.encode()).hexdigest()
+
+        output_folder = os.path.join(kangooroo_config["output_folder"], url_md5)
+        if not os.path.exists(output_folder):
+            possible_folders = os.listdir(kangooroo_config["output_folder"])
+            if len(possible_folders) == 0:
+                raise Exception(
+                    (
+                        "No Kangooroo output folder found. Kangooroo may have been OOMKilled. "
+                        "Check for memory usage and increase limit as needed."
+                    )
+                )
+            elif len(possible_folders) != 1:
+                raise Exception(
+                    (
+                        "Multiple Kangooroo output folders found. Unknown situation happened, you can try "
+                        "submitting this URL again to see if it would help."
+                    )
+                )
+            else:
+                url_hash_mismatch = ResultTextSection("URL hash mismatch", parent=request.result)
+                url_hash_mismatch.add_line(
+                    (
+                        f"URL '{request.task.fileinfo.uri_info.uri}' ({url_md5}) was requested "
+                        f"but a different URL was fetched ({possible_folders[0]})."
+                    )
+                )
+                output_folder = os.path.join(kangooroo_config["output_folder"], possible_folders[0])
+
+        results_filepath = os.path.join(output_folder, "results.json")
+        if not os.path.exists(results_filepath):
+            raise Exception(
+                (
+                    "No Kangooroo results.json found. Kangooroo may have been OOMKilled. "
+                    "Check for memory usage and increase limit as needed."
```

Comment:
I think this check should be done after the return statement. That way you return a single item, and you won't have to do another conditional check to add the file to the supplementary files.